### PR TITLE
Added constants list to gallery

### PR DESCRIPTION
**Pull request by ehsteve**
### Description

Adds an example to the gallery that simply lists all of the constants available similar to the colormap example.

### CODE REVIEW

**Comment by nabobalis** [comment]
You need to update your master.

File: examples/gallery/constants_reference.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+##############################################################################
+# All constants are stored in a dictionary. A list of all available keys
+# which describe the constant can be had with the following command.
+con.constants.keys()
+
+##############################################################################
+# The following command will display all constants as well as their values
+# in an astropy `Table <http://docs.astropy.org/en/stable/table/index.html>`_
+print(con.print_all())
```
Comment (dpshelio) [inline examples/gallery/constants_reference.py]: This `print` seems redundant.

File: examples/gallery/constants_reference.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+The example shows all of the constants provided by sunpy.
+"""
+
+from sunpy.sun import constants as con
+
+##############################################################################
+# All constants are stored in a dictionary. A list of all available keys
+# which describe the constant can be had with the following command.
+con.constants.keys()
```
Comment (dpshelio) [inline examples/gallery/constants_reference.py]: Whereas I would put a `print` in here.

**Comment by nabobalis** [comment]
Since the CI is on fire, the only build we care about is the docs one.

**Pull request merged**